vsutil/__init__.py

```diff
@@ -251,6 +251,65 @@ def is_image(filename: str, /) -> bool:
     return types_map.get(path.splitext(filename)[-1], '').startswith('image/')
 
 
+def scale(value: float, 
+          input_depth: int, 
+          output_depth: int, 
+          range_in: Union[int, Range] = 0, 
+          range: Optional[Union[int, Range]] = None, 
+          scale_offsets: bool = False, 
+          chroma: bool = False) -> float:
+    """
+    Scales a given value between bit depths, sample types, and/or ranges.
+    :value:         Numeric value to be scaled
+    :input_depth:   Bit depth of the "value" parameter. Use 32 for float samples
+    :output_depth:  Bit depth to scale the input value to
+    :range_in:      Pixel range of the input value. No clamping is performed
+    :range:         Pixel range of the output value. No clamping is performed
+    :scale_offsets: Whether or not to apply YUV offsets to float chroma and/or TV range integer values
+                    e.g. when scaling a TV range value of 16 to float, setting this to True will return "0.0" rather than "0.073059.."
+    :chroma:        Whether to treat values as chroma instead of luma
+    """
+
+    range_in = _resolve_enum(Range, range_in, 'range_in')
+    range = _resolve_enum(Range, range, 'range')
+    range = fallback(range, range_in)
+
+    if input_depth == 32:
+        input_peak = 1
+        range_in = 1
```

Comment:
I would prefer to set `range_in` in a separate block. Doing that means this if/elif chain would have no side effects other than setting `input_peak` which would allow us to extract the peak value logic into a small function to avoid the duplication.